### ISSUE TITLE
Test Watershed - Meherrin: MN4_8400_8380

### ISSUE THREAD

**Issue by juliabruneau** [HARPgroup/HARParchive#463]
**Running hsp2 for the southern basin Meherrin (MN4_8400_8380):**

Running the meteorology:
```
segs=`cbp get_landsegs MN4_8400_8380`
for i in $segs; do
  # convert grid CSVs into land segment CSVs
  ./a2l_one 1984010100 2020123123 /backup/meteorology/out/grid_met_csv /backup/meteorology/out/lseg_csv $i
  # update long term averages
  LongTermAvgRNMax /backup/meteorology/out/lseg_csv/1984010100-2020123123 /backup/meteorology/out/lseg_csv/RNMax 1 $i
  # finally, create a WDM for each land seg
  # this script reads the file /etc/hspf.config to get directories.
  ./wdm_pm_one $i 1984010100 2020123123 nldas2 harp2021 nldas1221 p20211221
done
```
Commands in order:
`cbp run_lug.csh p532sova_2021 MN4_8400_8380`

`cbp run_land.csh p532sova_2021 MN4_8400_8380`

`cbp run_rug.csh hsp2_2022 MN4_8400_8380`

`cbp run_river.csh hsp2_2022 MN4_8400_8380`

**Comment by juliabruneau** [comment]
### 1st run - errors

After run_land:
``` 
Error in FILES block - Return code 2
land segment:  A51111  did not run for land use:  afo
  input file  ../../..//tmp/uci/land/afo/p532sova_2021/afoA51111.uci
check the file ../../tmp/scratch/temp581411/afoA51111.ech
Warning: could not find /opt/model/p53/p532c-sova/run_bhatt in tree
```
After run_etm:
```
juliasb@deq2:/opt/model/p53/p532c-sova$ cbp run_etm.csh p532sova_2021 MN4_8400_8380
Loading /opt/model/p53/p532c-sova/hspf.config
Trying: p532sova_2021, MN1_7990_8100, [random.ksh] | ../../..//code/bin/make_binary_transfer_coeffs.exe
 Binary ETM file for MN1_7990_8100 p532sova_2021
   reading data files
   all data read, make binary etm file
A51037 for hvf hwm hom alf hyw hyo pas trp urs npd rcn afo nid
A51111 for

 PROBLEM FILE WRITTEN

 Error: problem opening wdm =           -2
../../../tmp/wdm/land/for/p532sova_2021/forA51111.wdm
opening to calculate return frequency

Trying: p532sova_2021, MN1_7990_8100, [random.ksh] | ../../..//code/bin/make_binary_transfer_coeffs.exe
 Binary ETM file for MN1_7990_8100 p532sova_2021
   reading data files
   all data read, make binary etm file
A51037 for hvf hwm hom alf hyw hyo pas trp urs npd rcn afo nid
A51111 for

 PROBLEM FILE WRITTEN

 Error: problem opening wdm =           -2
../../../tmp/wdm/land/for/p532sova_2021/forA51111.wdm
opening to calculate return frequency

Warning: could not find /opt/model/p53/p532c-sova/run_bhatt in tree
```

**Comment by rburghol** [comment]
OK - did run_lug.csh proceed OK?  Perhaps we need to get the meteorology in place, though I thought I had done that already.

FYI - once one level fails, such as run_land.csh in this case, no reason to go forward since we know the next level will be destinted to fail.

**Comment by juliabruneau** [comment]
> OK - did run_lug.csh proceed OK? Perhaps we need to get the meteorology in place, though I thought I had done that already.
> 
> FYI - once one level fails, such as run_land.csh in this case, no reason to go forward since we know the next level will be destinted to fail.

Yes run_lug was successful.

**Comment by juliabruneau** [comment]
### 2nd try

* After running the meteorology, all the run commands were successful until the last command:

```
juliasb@deq2:/opt/model/p53/p532c-sova$ cbp run_river.csh hsp2_2022 MN4_8400_8380
Loading /opt/model/p53/p532c-sova/hspf.config
********************************************
********************************************
********** HSPF Model Run Beginning ********
********************************************
********************************************
Found segments: MN1_7990_8100 MN2_8250_8190 MN2_8100_8190 MN3_8190_8260 MN4_8260_8400 MN4_8400_8380
Problem in river
 could not find eos wdm:
../../..//tmp/wdm/river/hsp2_2022/eos/MN1_7990_8100.wdm
 check that etm ran for segment  MN1_7990_8100
********************************************
********************************************
********** HSPF Model Run Beginning ********
********************************************
********************************************
Found segments: MN1_7990_8100 MN2_8250_8190 MN2_8100_8190 MN3_8190_8260 MN4_8260_8400 MN4_8400_8380
Problem in river
 could not find eos wdm:
../../..//tmp/wdm/river/hsp2_2022/eos/MN1_7990_8100.wdm
 check that etm ran for segment  MN1_7990_8100
Warning: could not find /opt/model/p53/p532c-sova/run_bhatt in tree
```

**Comment by rburghol** [comment]
@glenncampagna @juliabruneau I am pretty sure that the `eos` copy of the WDM is created during the ETM step.  Can you run that and see if there are problems?  I am thinking that the ETM step might require us to run for the `hsp2_2022` scenario, now that we are doing this new method with the downstream flow, or, maybe we always needed to run both ETMs?  Or maybe we *never* needed to run ETM for `p532sova_2021`, and instead should have run`hsp2_2022` all along?

**Comment by juliabruneau** [comment]
Running run_etm again (@rburghol )

We were successful in running run_etm with p532sova_2021 and hsp2_2022. p532sova_2021 was used initally, and now I ran hsp2_2022.

Then I ran the rest of the commands and **run_river ran successfully.**

But I noticed this mistake in the code with hsp_hydr_analysis Rscript:
```
Notice(analyze): Rscript /opt/model/p53/p532c-sova/run/export/hsp_hydr_analysis.R MN2_8100_8190 hsp2_2022 /media/model/p532/out/river/hsp2_2022/hydr/

Warning message:
In dir.create(file.path(image_directory_path)) :
  '/media/model/p532/out/river/hsp2_2022/hydr//images' already exists
Error in fread(hydr_file_path) :
  File '/media/model/p532/out/river/hsp2_2022/hydr//hydr/MN2_8100_8190_hydr.csv' does not exist or is non-readable. getwd()=='/opt/model/p53/p532c-sova/tmp/scratch/temp656570'
Execution halted
```
In the argument filepath /hydr/ is included, which is not supposed to be there since now in the image file path for example, there are two (/hydr//hydr/) in a row. Or it's causing two // in the filepath.

**Comment by rburghol** [comment]
Good catch.  That is easy to fix.  But it makes me wonder, why are we guessing the `hydr` sub-directory in that script?  Is there a good reason for that, as in, it needs to look in the base scenario directory?

I am asking as I think we just need to make sure that we are consistent.  Part of me thinks like you all did when making the script guess `hydr`, that is, we only give the base directory, then let the individual routine decide where inside that directory to put things.  However, we are already telling it to descend into river and hsp2_2022... so, 

Plus, we have another step that appears hard-wired in the Rscript, where it tries to create the directory `hsp2_2022/hydr`, so that is definitely *not* using the directory that we passed in.  If it *did*, we wouldn't get the error on `hydr/hydr` since it would have created it already. 

In short, I think our analysis scripts should either:
1. Guess all path info except the absolute minimum (`CBP_EXPORT_DIR`), 
2. or, should guess *none* of the info, that is, be given the full path to the output directory and only be responsible for creating files in that directory.

What are your thoughts in terms of how we've structured the scripts thus far?  Didn't you just write a script to help us get the output file path for something(please remind me)? My thinking is that the path script you recently created tells me that we prefer the 2nd choice -- but since you have been working on most scripts you have a better idea of what we've actually been doing, and where the 1st option might be superior to the 2nd.

**Comment by juliabruneau** [comment]
@rburghol 

>What are your thoughts in terms of how we've structured the scripts thus far? 
>
Initially when we created the scripts, we tried to make the arguments as 'user friendly' as possible. Now in the automation phase we understand how this might not be the best course of action in the long run. This is why we agree with having the more specific file paths as the arguments for the Rscripts.

>Didn't you just write a script to help us get the output file path for something(please remind me)?
>
Yes, we created a script/function to find the filepath to the h5 files.

We agree with option 2, those are the changes we implemented this week to the arguments in the batch_harvest_river, but didn't know we were going to work with the analysis script as well. 

We will implement changes to the **hsp_hydr_analysis** so that the entire file path will be the argument (option 2), and let you know of the updated use.

**Comment by glenncampagna** [comment]
### Updated use for hsp_hydr_analysis.R
Use: `hsp_hydr_analysis.R [riverseg] [scenario] [input file path] [image directory path]`
Example: `Rscript HARParchive/HARP-2022-Summer/AutomatedScripts/hsp_hydr_analysis.R JL1_6770_6850 hsp2_2022 /media/model/p532/out/river/hsp2_2022/hydr/ /media/model/p532/out/river/hsp2_2022/images/`

Exports: located at http://deq1.bse.vt.edu/d.dh/om-model-info/6856191 under the feature for JL1_6770_6850 with the vahydro ftype

**Comment by rburghol** [comment]
Thanks all - I concur with the decision for option 2.

**Comment by glenncampagna** [comment]
### Comparing hspf and hsp2 for MN4_8400_8380 
`"Avg % difference: 1.9019702"`
1. 
![image](https://user-images.githubusercontent.com/104520197/185410614-c55e78be-e1c4-41b7-89b4-9989eea699a1.png)

2. 
![image](https://user-images.githubusercontent.com/104520197/185411368-4a183183-5ad7-4711-8c17-3b064848e6cc.png)

**Comment by rburghol** [comment]
Can you all check to see if there are withdrawals in the HSPF version?  Using wdm2text the UCI will have DSNs for DIVR and DIVA - that might explain this constant difference?  I can also look into the landuse, but I *thought* they use identical land use input configurations.

**Comment by glenncampagna** [comment]
### Unable to retrieve DSNs 3007 and 3008 with wdm2text (DIVR & DIVA)
DIVR (3007):
```
/tmp/wdm/river/p532sova_2021/stream$ wdm2text
  program to write hourly ascii output from a wdm
 wdm name, start year, end year, dsn
MN4_8400_8380.wdm, 1984, 2020, 3007
$ cat problem 
 
 PROBLEM FILE WRITTEN
 
 error getting start and end of existing data                   
 error code=         -81                                        
 dsn =         3007 
```                                            
DIVA (3008):
```
/tmp/wdm/river/p532sova_2021/stream$ wdm2text
  program to write hourly ascii output from a wdm
 wdm name, start year, end year, dsn
MN4_8400_8380.wdm, 1984, 2020, 3008
$ cat problem 
 
 PROBLEM FILE WRITTEN
 
 error getting start and end of existing data                   
 error code=         -81                                        
 dsn =         3008                                             

```
